refactor: Remove unused E2E test helpers

dcalhoun opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->
Remove unused E2E test helpers.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->
Relates to https://github.com/WordPress/gutenberg/issues/55251. Reduce code complexity. These can be reinstated from 
Git history when beneficial.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->
- Remove unused `waitIfAndroid` utility
- Remove unused `clearClipboard` utility
- Remove unused `setClipboard` utility

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->
n/a

### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->
n/a

## Screenshots or screencast <!-- if applicable -->
n/a

dcalhoun committed:
test: Remove unused `waitIfAndroid` utility

dcalhoun committed:
test: Remove unused `clearClipboard` utility

dcalhoun committed:
test: Remove unused `setClipboard` utility

geriux commented on packages/react-native-editor/__device-tests__/helpers/utils.js:
```diff
@@ -682,42 +682,6 @@ const clickIfClickable = async (
- * @param {Object}               driver      Driver
- * @param {ClipboardContentType} contentType Type of the content
- */
-const clearClipboard = async ( driver, contentType = 'plaintext' ) => {
```
I think we'd need to remove its usage from [this test](https://github.com/WordPress/gutenberg/blob/1f907579e2cd6b580fd4bbf32a31e9f2f8877de9/packages/react-native-editor/__device-tests__/gutenberg-editor-drag-and-drop.test.js#L10).

dcalhoun committed:
test: Remove unnecessary `clearClipboard` usage

This became unnecessary when the clipboard-related tests were removed in
https://github.com/WordPress/gutenberg/pull/55541.

geriux reviewed (approved):
LGTM! Thank you for removing these 🚀
